### ISSUE TITLE
ImportError for Python version <3.8

### ISSUE THREAD

**Issue by hashstat** [inyutin/aiohttp_retry#7]
When importing aiohttp_retry 1.1 in Python 3.7 or lower an ImportError exception is raised while importing Protocol from the typing module. Protocol was added to the typing module in Python 3.8. For earlier versions, it needs to be imported from the typing_extensions module.

```
>>> import aiohttp_retry
../venv/lib/python3.7/site-packages/aiohttp_retry/__init__.py:6: in <module>
    from typing import Any, Callable, Optional, Set, Type, Protocol
ImportError: cannot import name 'Protocol' from 'typing' (/usr/local/lib/python3.7/typing.py)
```

**Comment by inyutin** [comment]
Thank you for finding such a nasty bug. 

I fixed it at v1.1.5. As I explain at https://github.com/inyutin/aiohttp_retry/issues/3 you probably should update to v2.0. However, it hasn't backward compatibility. If you want, you can use v1.2, it should be fine.

**Comment by hashstat** [comment]
I'm getting errors with v1.2 as well. We'll just pin the requirement at 1.0 until we have a chance to upgrade to v2.0.

**Comment by inyutin** [comment]
Strange, I'm tested both on python3.6 and python3.7. Can you, please, provide certain version of your python interpreter (python -V)  ?

On v1.2 there are should be only deprecated errors.

**Comment by hashstat** [comment]
This new error is my own fault. I'm overriding `aiohttp_retry.RetryClient._request()` to get some additional behaviors. Apparently the signature of that method changed and will require some additional changes on our end. That's what I get for messing with non-public members.

**Comment by inyutin** [comment]
Ok, I will close this issue then.

However, I would like to discuss why you need to override _request. Probably, if you describe why you need such a thing, I will change code to suit your needs. Open new issue if you want to discuss this thing.

**Comment by hashstat** [comment]
I'm not sure the features we added fit the scope of this project. We were moving from requests and implementing some things formerly accomplished with adapters, which aiohttp lacks. We added support to automatically call `raise_for_status()`, to disallow requests not starting with 'https://', and support for default timeout and headers. If any of these seem like features aiohttp_retry should support, I would be happy to share code.
